docs/demos/search/template_matching.jl

```diff
@@ -0,0 +1,63 @@
+# ---
+# title: Template Matching
+# cover: assets/template_matching.png
+# author: Michael Pusterhofer
+# date: 2021-06-22
+# ---
+
+# This demo shows how to find objects in an image using template matching. 
+
+# The main idea is to check the similarity between a search target(template) and a subsection of the image. The subsection is usually the same size as the template. Therefore by using [`mapwindow`](@ref) we can assign a value for every subsection of the image.
+
+# At first we import the following packages. 
+using ImageCore: Gray
+using ImageMorphology: label_components, component_centroids
+using ImageFiltering: mapwindow, Fill, imfilter, KernelFactors
+using ImageDistances: sqeuclidean
+using ImageContrastAdjustment: adjust_histogram, LinearStretching
+using TestImages
+using Plots: scatter!, plot
+
+# `ImageCore` enables the generation of images, `ImageFiltering` provides the [`mapwindow`](@ref) function and `ImageFeatures` provides functions to label segments of an image. To calculate the similarity `squeclidean` is used. This defines a distance between two images as:
+#
+# ```math
+# distance = \sum_i^[number of pixels} (template_i - subsection_i)^2
+# ```
+#
+# `ImageContrastAdjustment` provides functions to adjust the histogram, which is useful when an image contains values bigger than 1 or smaller than 0.  The `Testimages` package will provide our image and plot is used to overlay a scatter plot onto the image.
+
+# To start we first load our image.
+img = testimage("lena_gray_512")
```

Comment:
For some reason, we don't recommend to use lena in new codes (https://github.com/JuliaImages/TestImages.jl/issues/72).

Reply:
Ok, I'll try to use another one.